functions/src/modules/SQR.functions.ts

```diff
@@ -512,4 +512,16 @@ export const exportSubmissionsToSpreadsheet = functions.database
         'Email Address': withDefault(author.emailAddress),
       });
     }
-  );
+  )
+
+export const processSQRDoneFromGmail = async (processObject: any) => {
+  const listName = processObject.fileName.split("-")[0];
+  const soundQualityReporting = db.ref(`files/${listName}/${processObject.fileName}/soundQualityReporting`);
+  const sqrResults = await soundQualityReporting.once('value');
+  if (sqrResults.exists()) {
+    await soundQualityReporting.update({
+      status: "Done",
+      timestampDone: moment(parseInt(processObject.lastThreadMessageTimestamp)).unix(),
```

Comment:
Isn't `processObject.lastThreadMessageTimestamp` already a timestamp?

Reply:
I think I did this because it's timestamp in milliseconds, and we are using epoch in seconds in other places. And you said to use `moment` to make it more clear. I guess we can just use as is though.

Reply:
If you are absolutely sure that `processObject.lastThreadMessageTimestamp` is in milliseconds and that momentjs handles it correctly, then okay. I'm just doubting.

Reply:
Yep, `internalDate` is in seconds: https://developers.google.com/gmail/api/v1/reference/users/messages

https://momentjs.com/docs/#/parsing/unix-timestamp-milliseconds
`moment()` takes milliseconds and `unix()` gives back seconds.